### ISSUE TITLE
Trainer always causes divergence after `Trainer.tuner.lr_find`

### ISSUE THREAD

**Issue by Callidior** [Lightning-AI/lightning#12670]
## 🐛 Bug

We determine the learning rate using Lightnings LR finder as follows:

```python
trainer = pl.Trainer(max_epochs=16, gpus=1, auto_lr_find=True)
lr_finder = trainer.tuner.lr_find(
    model, train_dataloaders=train_loader,
)
lr = lr_finder.suggestion() or 0.001
```

Then, we re-instantiate the module with the new learning rate and call `trainer.fit()` _using the same Trainer instance_ as for finding the learning rate.

In PyTorch Lightning 1.5.x, this worked without issues. In Lightning 1.6, this always leads to divergence, regardless of the learning rate. Using a new Trainer instance works, but from my understanding, the trainer should still be usable after having called `trainer.tuner.lr_find()`.

### To Reproduce

The following Colab reproduces the issue with the Boring Model: https://colab.research.google.com/drive/12BF8m4bCe6ijWDQ1KCEXr5wNAL0TCyhW?usp=sharing

Changing the pytorch_lightning version in the first cell to 1.5.10 solves the issue.

### Expected behavior

Calling `trainer.fit()` after `trainer.tuner.lr_find()` works just like calling `trainer.fit()` right after instantiating the trainer.

### Environment

* CUDA:
	- GPU:
		- Tesla K80
	- available:         True
	- version:           11.1
* Packages:
	- numpy:             1.21.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.10.0+cu111
	- pytorch-lightning: 1.6.0
	- tqdm:              4.63.0
* System:
	- OS:                Linux
	- architecture: 64bit
	- processor:         x86_64
	- python:            3.7.13

cc @akihironitta @borda @rohitgr7

**Comment by NikolasWolke** [comment]
I noticed the same problem. During `lr_find` the learning rate scheduler is replaced by an exponential schedule and the trainer still has a reference to that scheduler (see `trainer.lr_schedulers`). So when you start the training with the same trainer instance you should see the learning rate diverging quickly.

**Comment by Callidior** [comment]
As far as I know, this issue has not been resolved yet, and it basically breaks the LR finding feature.

**Comment by faroit** [comment]
@NikolasWolke seems like a really bug. What is the simpliest workaround?

**Comment by krishnakalyan3** [comment]
@Callidior I was unable to reproduce this issue. I used the[ hello world MNIST](https://github.com/Lightning-AI/tutorials/blob/main/lightning_examples/mnist-hello-world/hello-world.py) example to reproduce. Things work as expected.
Can you please retry your experiments on a dataset like (imagenet, cifar, etc) . I guess the boring model uses random dataset and I guess that might the root cause to the divergence you see.

<img width="1039" alt="image" src="https://user-images.githubusercontent.com/4648995/185095525-fb9cc59e-5170-4050-85b6-3f7d8c27d265.png">

#14113

**Comment by Callidior** [comment]
@krishnakalyan3 We did also observe the issue on our use-cases with real data, not just with the boring model. The actual problem does not occur on the call to `lr_find`, but when you try to re-use the trainer instance afterwards, e.g., for actually training your model, which would be the logical next step.

Since Lightning version 1.6, the LR finder does not reset the trainer properly, so that it will still use the LR scheduler used by the LR finder for trying increasing learning rates. This results in quick divergence.

A simple workaround is to create a new trainer instance. But I assume it should be possible to re-use the same trainer. Otherwise, it should be documented that the trainer is broken after `lr_find()`.

I verified that the fix in #14113 resolves this issue. I also reproduced it on MNIST, as requested: https://colab.research.google.com/drive/1rPPOh_RjGEw9D_V7k57OhmbhtxUjmZ_t?usp=sharing
Note that the loss becomes NaN already after the first epoch of training.

![image](https://user-images.githubusercontent.com/7915048/185101347-35f01cdc-2477-4c73-bd41-b47c31557758.png)
![image](https://user-images.githubusercontent.com/7915048/185101400-e76445e7-4e87-45ed-a2f8-b069be95a37d.png)
